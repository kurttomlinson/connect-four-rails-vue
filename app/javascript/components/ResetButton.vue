<template>
  <div class="centered">
    <button @click="$emit('resetGame')">Reset</button>
  </div>
</template>

<script>

export default {
  methods: {
    resetGame () {
      this.$emit('resetGame')
    }
  }
}
</script>

<style scoped>
.centered {
  text-align: center;
}
</style>
