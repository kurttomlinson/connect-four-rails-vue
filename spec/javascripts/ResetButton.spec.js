import { shallow } from 'vue-test-utils'
import Vue from 'vue'
import ResetButton from 'components/ResetButton'

describe('ResetButton.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(ResetButton)
  })

  it('emits a "resetGame" event when clicked', () => {
    const stub = jest.fn()
    wrapper.vm.$on('resetGame', stub)
    const button = wrapper.find('button')
    button.trigger('click')
    // Vue.nextTick(function () {
      expect(stub).toBeCalled()
    // })
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})