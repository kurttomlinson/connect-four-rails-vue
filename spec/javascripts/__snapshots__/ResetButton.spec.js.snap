// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetButton.vue has the expected html structure 1`] = `
<div
  class="centered"
>
  <button>
    Reset
  </button>
</div>
`;
